src/boost-api/Endpoint_Handlers.cpp

```diff
@@ -143,6 +131,68 @@ namespace api {
         }
     }
 
+    void handle_knn_query(const request<string_body> &req, response<string_body> &res){
+        try {
+            boost::json::value json_data = get_json_from_request_body(req, res);
+
+            if (!json_data.is_object()) {
+                throw std::runtime_error("Invalid JSON data: expected an object");
+            }
+
+            const boost::json::object &json_object = json_data.as_object();
+
+            std::string table_key{};
+            if (json_object.contains("db_table")) {
+                table_key = json_object.at("db_table").as_string().c_str();
+            } else {
+                throw std::runtime_error("JSON object does not contain 'db_table'");
+            }
+
+            std::string db_table{};
+            if (table_key == "original") {db_table = "original_trajectories"; }
+            else if (table_key == "simplified") {db_table = "simplified_trajectories"; }
+            else { throw std::runtime_error("Error in db_table, must be either 'simplified' or 'original' "); }
```

Comment:
Du kan nøjes med at definiere `db_table` uden for din if-statement på linje 144 og så håndtere det hele inden i den første if-statement.